---
title: C26111 |Microsoft 文件
ms.custom: ''
ms.date: 11/04/2016
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-ide-code-analysis
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
caps.latest.revision: 8
author: mikeblome
ms.author: mblome
manager: ghogen
ms.workload:
- multiple
ms.openlocfilehash: dadc0c15e07595d3a35d4cb4a92f257bdcdf3490
ms.sourcegitcommit: 32f1a690fc445f9586d53698fc82c7debd784eeb
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 12/22/2017
---
# <a name="c26111"></a>C26111
警告 C26111： 無法釋放鎖定的呼叫端\<鎖定 > 函式的呼叫之前\<func >。  
  
 註解`_Requires_lock_not_held_`會加諸函式呼叫時指定的鎖定的鎖定計數不能小於或等於零的前置條件。 當函式無法釋出鎖定，然後呼叫另一個函式時，會發出警告 C26111。  
  
## <a name="example"></a>範例  
 下列範例會產生警告 C26111 因為`_Requires_lock_not_held_`前置條件由呼叫違反`DoNotLock`鎖定的區段內。  
  
```  
  
typedef struct _DATA   
{  
    CRITICAL_SECTION cs;  
    int d;  
} DATA;  
  
_Requires_lock_not_held_(p->cs)   
  
void DoNotLock(DATA* p)  
{  
    EnterCriticalSection(&p->cs);   
    p->d = 0;  
    LeaveCriticalSection(&p->cs);  
}  
  
void LockedFunction(DATA* p)   
{   
    EnterCriticalSection(&p->cs);   
    DoNotLock(p); // Warning C26111  
    LeaveCriticalSection(&p->cs);  
}  
  
```